src/main/java/net/minecraftforge/registries/VanillaRegisterEvent.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Minecraft Forge - Forge Development LLC
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.registries;
+
+import net.minecraft.core.Registry;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.fml.event.IModBusEvent;
+import org.jetbrains.annotations.NotNull;
+
+class VanillaRegisterEvent extends Event implements IModBusEvent
```

Comment:
Any reason this is package protected instead of public? An event like this would be useful outside deferred registers for registering things into vanilla registries using the classic event style.

Also, could this be made a generic event? That way you could subscribe to `VanillaRegisterEvent<Recipe type<?>` for instance.

Reply:
The entire concept is that we can no longer garentee the type. The plan is to remove types all together in 1.19.
And the reason it is private is because We do not wish modders to use it. In 1.19 the Register event will become non-generic, and instead be keyed on Registry Name.

This event is purely a stop gap until we can make that breaking change.